.github/workflows/cut-release-branch.yaml

```diff
@@ -0,0 +1,283 @@
+
+# TODOs
+# add note to eventually commit changes directly and bypass checks - same as release - when we move to this model run test action after merge
+
+name: Cut new release branch
+
+on:
+  workflow_call:
+    inputs:
+      version_to_bump_main:
+        required: true
+        type: string
+      new_branch_name:
+        description: 'The full name of the new branch (ex. 1.5.latest)'
+        required: true
+        type: string
+      repository:
+        required: true
+        type: string
+      PR_title:
+        required: false
+        type: string
+      PR_body:
+        required: false
+        type: string
+    outputs:
+      PR_number:
+        description: "PR number for the opened PR"
+        value: ${{ jobs.cleanup_main.outputs.pr_number }}
+    secrets:
+      FISHTOWN_BOT_PAT:
+        description: "PAT that allows commits, creating branches and opening PRs so the repository"
+        required: true
+
+  workflow_dispatch:
+    inputs:
+      version_to_bump_main:
+        description: 'The alpha version main should bump to (ex. 1.6.0a1)'
+        required: true
+      new_branch_name:
+        description: 'The full name of the new branch (ex. 1.5.latest)'
+        required: true
+      repository:
+        description: "Repository to create new branch and cleanup main:"
+        required: true
+        type: choice
+        options:
+        - 'dbt-core'
+        - 'dbt-snowflake'
+        - 'dbt-redshift'
+        - 'dbt-bigquery'
+        - 'dbt-spark' 
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: write
+
+jobs:
+  prep_work:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "[DEBUG] Print Inputs"
+        run: |
+          echo "Version to bump main:    ${{ inputs.version_to_bump_main }}"
+          echo "Name of branch to cut:   ${{ inputs.new_branch_name }}"
+
+      - name: Audit New Version and Parse Into Parts
+        id: semver
+        uses: dbt-labs/actions/parse-semver@v1
+        with:
+          version: ${{ inputs.version_to_bump_main }}
+
+  cut_branch:
+    needs: ['prep_work']
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: "Checkout ${{ inputs.repository }} repository"
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ inputs.repository }}
+          ref: "main"
+          token: ${{ secrets.FISHTOWN_BOT_PAT }}
+
+      - name: "Ensure New Branch Does Not Exist"
+        id: check_new_branch
+        run: |
+          if git show-ref --quiet refs/heads/${{ inputs.new_branch_name }}; then
+            echo "Branch ${{ inputs.new_branch_name }} already exists.  Exiting."
+            exit 1
+          fi
+      
+      - name: "Create New Release Branch"
+        run: |
+          git checkout -b ${{ inputs.new_branch_name }}
+
+      - name: "Bump dependencies via script"
+        if: repo != dbt-core
+        run: |
+          echo "running some script yet to be written now"
+          scripts/update_dependencies.sh
```

Comment:
Is this something we have for the adapter repos?

Reply:
@leahwicz it's not something we have yet.  It will need to be written as part of implementation.  My plan was to get it working for core and then work on the script for adapter repos.